presentation/src/main/java/me/androidbox/presentation/login/viewmodel/LoginViewModel.kt

```diff
@@ -0,0 +1,33 @@
+package me.androidbox.presentation.login.viewmodel
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor() : ViewModel() {
+
+    var username by mutableStateOf("")
+        private set
+
+    var password by mutableStateOf("")
+        private set
+
+    var isPasswordVisible by mutableStateOf(false)
+        private set
```

Comment:
I personally like to keep my state in just one dataclass, since it makes it a bit more easier and cleaner

Reply:
@philipplackner @MKiperszmid 
For states I am wondering should we be using savedStatehandle with stateflow?

The reason I used mutableStateOf as its tried more closely to compose